actors/builtin/miner/miner_test.go

```diff
@@ -1,12 +1,160 @@
 package miner_test
 
 import (
+	"bytes"
+	"context"
 	"testing"
 
+	addr "github.com/filecoin-project/go-address"
+	"github.com/minio/blake2b-simd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
 	"github.com/filecoin-project/specs-actors/actors/builtin/miner"
+	"github.com/filecoin-project/specs-actors/actors/builtin/power"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
 	"github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
 )
 
+const SectorSize = abi.SectorSize(32 << 20)
+
 func TestExports(t *testing.T) {
 	mock.CheckActorExports(t, miner.Actor{})
 }
+func TestConstruction(t *testing.T) {
+	actor := miner.Actor{}
+
+	owner := tutil.NewIDAddr(t, 100)
+	worker := tutil.NewIDAddr(t, 101)
+	workerKey := tutil.NewBLSAddr(t, 0)
+	receiver := tutil.NewIDAddr(t, 1000)
+	builder := mock.NewBuilder(context.Background(), receiver).
+		WithActorType(owner, builtin.AccountActorCodeID).
+		WithActorType(worker, builtin.AccountActorCodeID).
+		WithHasher(blake2b.Sum256).
+		WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple construction", func(t *testing.T) {
+		rt := builder.Build(t)
+		params := miner.ConstructorParams{
+			OwnerAddr:  owner,
+			WorkerAddr: worker,
+			SectorSize: SectorSize,
+			PeerId:     "peer",
+		}
+
+		provingPeriodBoundary := abi.ChainEpoch(2386) // This is just set from running the code.
```

Comment:
Since the runtime's `Hasher` is swappable we _could_ pass a mockHasher that returns a value we define, but this seems reasonable enough.